src/components/eip4626/user/VaultInfo.tsx

```diff
@@ -17,41 +42,108 @@ export function VaultInfo({ vaultAddress }: VaultInfoProps) {
     vaultFeeTokenAddress = vaultFeeConfig[1]?.toString();
     vaultFeePercentage = vaultFeeConfig[2]?.toString();
-  }
 
-  if (vaultFeePercentage) {
-    vaultFeePercentageFormatted = Number(vaultFeePercentage) / 100;
+    if (vaultFeePercentage) {
+      vaultFeePercentageFormatted = Number(vaultFeePercentage) / 100;
+    }
   }
 
+  const renderInfoBlock = (blockKey: VaultInfoBlock) => {
+    let _value;
+
+    if (blockKey === "vaultAddress") {
+      _value = vaultAddress;
+    } else if (blockKey === "rewardAssets") {
+      _value = (
+        <>{rewardAssets?.map((asset) => <Text key={asset}>{asset}</Text>)}</>
+      );
+    } else if (blockKey === "totalAssets") {
+      _value = formatBalance(vaultAssetTotalAssets?.toString());
+    } else if (blockKey === "vaultAssetAddress") {
+      _value = vaultAssetAddress;
+    } else if (blockKey === "vaultShareAddress") {
+      _value = vaultShareAddress;
+    } else if (blockKey === "vaultFees") {
+      _value = (
+        <>
+          {vaultFeeReceiver && (
+            <Text>Vault fee receiver address: {vaultFeeReceiver}</Text>
+          )}
+          {vaultFeeTokenAddress && (
+            <Text>Vault fee token address: {vaultFeeTokenAddress}</Text>
+          )}
+          {vaultFeePercentageFormatted && (
+            <Text>Vault fee percentage: {vaultFeePercentageFormatted}%</Text>
+          )}
+        </>
+      );
+    }
+
+    if (blockKey === "vaultFees") {
+      return (
+        <>
+          {vaultFeeReceiver && (
+            <Flex direction="row" gap="1" justify="space-between">
+              <Text fontSize={14} fontWeight="800" width="45%">
+                Vault fee receiver address
+              </Text>
+              <Text fontSize={14} width="64%">
+                {vaultFeeReceiver}
+              </Text>
+            </Flex>
+          )}
+          {vaultFeePercentageFormatted && (
+            <Flex direction="row" gap="1" justify="space-between">
+              <Text fontSize={14} fontWeight="800" width="45%">
+                Vault fee percentage
+              </Text>
+              <Text fontSize={14} width="64%">
+                {vaultFeePercentageFormatted}%
+              </Text>
+            </Flex>
+          )}
+          {vaultFeeTokenAddress && (
+            <Flex direction="row" gap="1" justify="space-between">
+              <Text fontSize={14} fontWeight="800" width="45%">
+                Vault fee token address
+              </Text>
+              <Text fontSize={14} width="64%">
+                {vaultFeeTokenAddress}
+              </Text>
+            </Flex>
+          )}
+        </>
+      );
+    }
+
+    return (
+      <Flex direction="row" gap="1" justify="space-between">
+        <Text fontSize={14} fontWeight="800" width="45%">
+          {vaultInfoBlockPlaceholders[blockKey]}
+        </Text>
+        <Text fontSize={14} width="64%">
+          {_value}
+        </Text>
+      </Flex>
+    );
+  };
```

Comment:
I'd like to suggest a better structure, easier to manage and support.

- make a generic `VaultInfoContainer` component, which would show the passed title and children, smth like
```
 return (
      <Flex direction="row" gap="1" justify="space-between">
        <Text fontSize={14} fontWeight="800" width="45%">
          {title}
        </Text>
        <Text fontSize={14} width="64%">
          {children}
        </Text>
      </Flex>
    );
```

- make a separate component for each _value

- drop the iteration, just have one by one like this

```
<VaultInfoContainer title='Vault Address'>
<VaultInfoAddress />
</VaultInfoContainer>

<VaultInfoContainer title='Vault Fess'>
<VaultInfoFees />
</VaultInfoContainer>
... etc
```

Reply:
This approach improves reusability, structure and is easier to read and understand.